Added extended BLE commands.

idaniel86 opened the pull request:
Hi guys, there is an issue with legacy BLE commands using bluez with version 5 and probably higher. Commands like OCF_LE_SET_SCAN_PARAMETERS and OCF_LE_SET_SCAN_ENABLE return error COMMAND_DISALLOWED. After reading few discussions this is due to commands being deprecated and extended version of these commands should be used. I have added check for HCI version and based on this value use legacy or extended command versions. I had this issue on Ubuntu 18.04 and 19.10. When reverting the Ubuntu 18.04 kernel to the original one, the legacy commands worked. My update made beacontools to work with Ubuntu 19.10 and bluez 5.50. Feel free to rework the changes I proposed.

citruz commented on beacontools/scanner.py:
```diff
@@ -229,6 +318,55 @@ def process_packet(self, pkt):
                     self.callback(bt_addr, rssi, packet, properties)
                     return
 
+    def process_ext_packet(self, pkt):
```
Same here as for `set_ext_scan_parameters`. Please merge this into `process_packet` to avoid duplicate code. The only thing that seems to be different is the offset of `payload` and `rssi`.

citruz reviewed (changes requested):
Thanks for the pull request! This looks really important to make beacontools future proof.
Please see my comments for some changes that we should do before merging to avoid code duplication.
There are also some linting errors. See [https://travis-ci.org/github/citruz/beacontools/jobs/732906931](https://travis-ci.org/github/citruz/beacontools/jobs/732906931)

idaniel86 commented:
Merged the commands and added descriptions. However I am getting still a stupid lint error "R0913: Too many arguments (7/5) (too-many-arguments)" in the helper hci_send_req function that just calls the bluez hci_send_req function and passes down the arguments.

BTW: An important issue, I have implemented the hci_send_req only for linux backend! Someone with knowledge how to implement it under freebsd and has an environment for testing has to implement it.

citruz commented:
Sorry for the delay. I now have a setup to test your PR and will hopefully be able to look at it in the next week.
Regarding FreeBSD, maybe @myfreeweb can help?

citruz commented:
Awesome! Except for a [small typo](https://github.com/citruz/beacontools/pull/55/commits/da3e6460aa78f755bb7715c3c384a0937cde4414), your PR worked out of the box on my Ubuntu 20.04 test machine.
I added a `NotImplementedError` for `send_req` in the FreeBSD backend, so that it falls back to the legacy handling automatically. In that way we don't break the FreeBSD support completely.
I will finish the release in the next days. Thanks again for the contribution 🚀

citruz commented:
(This should also fix #46)

idaniel86 commented:
Perfect!!! I am glad I could be of help.
